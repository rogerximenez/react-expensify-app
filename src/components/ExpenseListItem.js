import React from "react";
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt}) => (
  
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <div>{amount}</div>
    <div>{createdAt}</div>
  </div>
  );

export default ExpenseListItem;

