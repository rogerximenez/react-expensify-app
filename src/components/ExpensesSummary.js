import React from "react";
import { connect } from "react-redux";
import { Link } from 'react-router-dom'
import numeral from "numeral";
import expensesTotal from "../selectors/expenses-total";
import selectExpenses from '../selectors/expenses'


export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWording = expensesCount === 1 ? 'expense' : 'expenses';
  const expenseTotalFormatted = numeral(expensesTotal / 100).format('$0,0.00');

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWording} totaling <span>{expenseTotalFormatted}</span></h1>
        <div className="page-header__actions">
          <Link className="btn-primary" to="/dashboard/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: expensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary)