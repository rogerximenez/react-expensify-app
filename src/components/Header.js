import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/dashboard/create" activeClassName="is-active">Create Expense</NavLink>
    <NavLink to="/dashboard/help" activeClassName="is-active">Help</NavLink>
  </header>
)

export default Header;