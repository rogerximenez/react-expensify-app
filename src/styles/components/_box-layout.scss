.box-layout {
  background:url('/images/dallas_background.jpg');
  background-size: cover;
  height: 100vh;
  width:  100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: fade-out(white, .15);
  border-radius: 6px;
  text-align: center;
  width: 30rem;
  padding: $xl-size $l-size;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}