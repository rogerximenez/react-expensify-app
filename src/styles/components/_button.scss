.btn-primary {
  background: $purple;
  color: white;
  border-radius: 4px;
  font-weight: 500;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
  font-size: $font-size-large;
  padding: $m-size $l-size;
  border: none;
  box-shadow: $box-shadow;
}

.btn-secondary {
  @extend .btn-primary;
  background: $very-light-gray;
  color: $dark-gray;
  padding: $s-size $m-size;
}

.btn-link {
  background: none;
  box-shadow: none;
  font-weight: 400;
}