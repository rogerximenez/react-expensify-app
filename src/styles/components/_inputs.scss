.text-input {
  box-shadow: $input-shadow;
  height: 50px;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
  border: 0.25px solid $light-gray;
  border-radius: 4px;
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}