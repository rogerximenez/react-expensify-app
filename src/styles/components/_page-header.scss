.page-header {
  background: $very-light-gray;
  margin-bottom: $l-size;
  padding: $l-size;
}

.page-header__title {
  font-weight: 300;
  margin: 0;
  span {
    font-weight: 700;
  }
}

.page-header__actions {
  margin-top: $m-size;
}