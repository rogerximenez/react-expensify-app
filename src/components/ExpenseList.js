import React from "react";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from '../selectors/expenses'
import { connect } from "react-redux";

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        props.expenses.map((expense) => {
        return <ExpenseListItem key={expense.id} {...expense} />
      })
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);