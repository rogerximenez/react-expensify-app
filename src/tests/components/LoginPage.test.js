import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";

test('should render the Login Page', () => {
  const wrapper = shallow(<LoginPage />)
  expect(wrapper).toMatchSnapshot();
})