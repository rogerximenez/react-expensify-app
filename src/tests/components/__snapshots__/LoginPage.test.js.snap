// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Login Page 1`] = `
<div>
  <h1>
    Log In To Your Account
  </h1>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
